### PR TITLE
Swerve Class documentation

### PR DESCRIPTION
**Pull request by 0x5b62656e5d**

### CODE REVIEW

**Comment by 0x5b62656e5d** [comment]
Links will need to be updated; WPI API is down

File: source/Docs/Software/SwerveClass.rst [hunk 1]
```diff
@@ -0,0 +1,205 @@
+##################
+Swerve Class
+##################
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:3]:
```suggestion
############
Swerve Class
############
```

File: source/Docs/Software/SwerveClass.rst [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+.. note:: 
+
+    For API specific issues, please see the `WPILib API <https://www.youtube.com/watch?v=dQw4w9WgXcQ>`_
+    , `CTRE API <https://api.ctr-electronics.com/phoenix/release/java/>`_, and the `REV Robotics API 
+    <https://codedocs.revrobotics.com/java/com/revrobotics/package-summary.html>`_.
+
+
+Pigeon2 Sensor
+==============
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:16]:
```suggestion
IMU Selection
=============
```

File: source/Docs/Software/SwerveClass.rst [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+The image below is a Pigeon 2.0 sensor.
+
+.. figure:: ../Photos/Software/Pigeon2_CTRE.webp
+    :scale: 30%
+    :alt: Pigeon 2.0 sensor
+
+mImu is the Pigeon2, or the Inertial Measurement Unit, which contains an accelerometer to sense and measure acceleration, a gyro to measure angular orientation, 
+and a 3 axis magnetometer to measure the magnetic waves around it. A WPI_Pigeon2 controller is used to call methods to get the values from this sensor.
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:25]: basically there are a lot more IMUs in FRC than just the pigeon 2. then list common choices of IMUs: Pigeon2, navX2, PigeonIMU, navX. make sure to include links to their API & product pages

File: source/Docs/Software/SwerveClass.rst [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+Constructor
+===============
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:28]:
```suggestion
===========
```

File: source/Docs/Software/SwerveClass.rst [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+.. code-block:: java
+
+    // Initialize IMU
+    private final WPI_Pigeon2 mImu = new WPI_Pigeon2(SwerveConstants.kImuID);
+
+    // Initializes the 4 swerve modules
+    private final SwerveModule mLeftFrontModule, mRightFrontModule, mLeftRearModule, mRightRearModule;
+    // Initializes odometry
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:37]:
```suggestion
    
    // Initializes odometry
```

File: source/Docs/Software/SwerveClass.rst [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    private SwerveDriveOdometry mOdometry;
+
+    public Swerve() {
+        // Initializes the 4 swerve modules with SwerveConstants
+
+        // First set it at zero, look at offset, then change
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:43]:
```suggestion

```

File: source/Docs/Software/SwerveClass.rst [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    public void periodic() {
+        mOdometry.update( //updates robot position and rotation
+            mImu.getRotation2d(), //gets robot rotation
+            getModuleStates()[0], 
+            getModuleStates()[1], 
+            getModuleStates()[2],
+            getModuleStates()[3]
+        );
+    }
+
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:96]: periodic shouldn't be rly considered as a method, maybe move it out of the method subcategory

**Review by jazonshou — CHANGES_REQUESTED** [review]
Good job so far, try to work on these suggestions for now. I'll look at it more later

File: source/Docs/Software/SwerveClass.rst [hunk 8]
```diff
@@ -12,20 +12,25 @@ This class will utilize the swerve module class to create the 4 objects used for
...
 
 .. figure:: ../Photos/Software/Pigeon2_CTRE.webp
     :scale: 30%
     :alt: Pigeon 2.0 sensor
 
-mImu is the Pigeon2, or the Inertial Measurement Unit, which contains an accelerometer to sense and measure acceleration, a gyro to measure angular orientation, 
-and a 3 axis magnetometer to measure the magnetic waves around it. A WPI_Pigeon2 controller is used to call methods to get the values from this sensor.
+mImu is the `Pigeon2 <https://store.ctr-electronics.com/pigeon-2/>`_ (`API <https://api.ctr-electronics.com/phoenix/release/java/com/ctre/phoenix/sensors/WPI_Pigeon2.html>`_)
+, or the Inertial Measurement Unit, which contains an accelerometer to sense and measure acceleration, a gyro to measure angular orientation, 
+and a 3 axis magnetometer to measure the magnetic waves around it. A WPI_Pigeon2 controller is used to call methods to get the values from this sensor.
```
Comment (jazonshou) [inline source/Docs/Software/SwerveClass.rst:26]: You dont need all this extra info - just specify that it's an imu & it measures angle of the robot

also specify that pigeon2 will be used in the docs

**Review by jazonshou — CHANGES_REQUESTED** [review]

**Review by jazonshou — APPROVED** [review]

**Pull request merged**